execute/plugin.go

```diff
@@ -285,6 +307,15 @@ func (p *Plugin) Observation(
 			for _, msg := range commitReport.Messages {
 				sender := typeconv.AddressBytesToString(msg.Sender[:], uint64(p.cfg.DestChain))
 				nonceRequestArgs[commitReport.SourceChain][sender] = struct{}{}
+				messages = append(messages, msg)
+			}
+		}
+
+		prices := exectypes.PriceObservations{}
+		if p.messageExecutionEconomicsEnabled {
+			prices, err = p.priceObserver.ObservePrices(messages)
```

Comment:
Is it really sufficient to just call this here? I believe the price observations are a mix of source and dest information, some nodes may have source info, some may have dest info, wouldn't we need to consensus-ify these values (median, etc.) and only then do the boosting on the medianized values?

Reply:
Or is this stuff going to be done in upcoming PRs?

Reply:
Medianizing would happen in the outcome phase based on these observations.

Reply:
> I believe the price observations are a mix of source and dest information

Minor point, but I believe everything should be on the dest chain.

> wouldn't we need to consensus-ify these values (median, etc.) and only then do the boosting on the medianized values?

We do this consensus in `mergePriceObservations` in `execute/plugin_functions.go`